src/Controller/Component/PublicationComponent.php

```diff
@@ -121,20 +121,36 @@ public function genericTreeAction(string $path = ''): Response
             $this->getController()->set(['children' => $object['children'], 'childrenByType' => $object['childrenByType']]);
         }
 
-        // Search template by uname, type or ancestors' uname
-        $templates = array_merge(
-            empty($path) ? ['home'] : [],
-            [$object['uname'], $object['type']],
-            array_map(
-                function (ObjectEntity $object): string {
-                    return $object['uname'];
-                },
-                array_reverse($ancestors)
-            ),
-            ['objects']
-        );
+        return $this->renderFirstTemplate(...$this->getTemplatesToIterate($object, ...array_reverse($ancestors)));
```

Comment:
The `...` operator already iterates throughout the `\Generator` returned by the `getTemplatesToIterate` method. Although it is not a big issue, we could probably avoid the tiny overhead of computing the object type's inherited types. However, this would require refactoring the `renderFirstTemplate` method's signature, which would add backward incompatibility.

Long story short, despite being sub-optimal, I'd leave this line as is and deal with the slight overhead.